src/rest/util.ts

```diff
@@ -37,15 +38,19 @@ export const handleError = (
 };
 
 /**
- * Create a new token for authentication. Requires the
- * `AUTH_SEED` environment variable to be set to obfuscate
- * the values of the tokens generated.
+ * Create a unique new token for authentication. Connects to the remote
+ * database.
  * @returns the token.
  */
-export const createToken = (): string => {
-	return createHash("sha256")
-		.update(`${new Date().toISOString()}${process.env.AUTH_SEED}`)
-		.digest("hex");
+export const createToken = async (): Promise<string> => {
+	const { _col, _connection } = await getCollection<User>("user");
+
+	let token: string;
+	do {
+		token = createHash("sha256").update(`${randomUUID()}`).digest("hex");
+	} while ((await _col.findOne({ token: token })) !== null); // loop until token is unique
```

Comment:
isn't this usage of do-while a little bit forced? could easily just initialize the variable when it's declared, and it would be less code

Reply:
I just didn't wanna type that expression for token twice but I can do that